### PR TITLE
Fix emulation/qemu-armv7 mainboard.

### PR DESCRIPTION
**Pull request by doriath**
This change fixes the rust code (broken by recent refactorings) and
fixes the fixed-dtfs layout to provide enough space for the actual
bootblob (it is over 6M but only 512K was reserved).

Ref #283

Signed-off-by: Tomasz Zurkowski <zurkowski@google.com>

### CODE REVIEW

File: src/mainboard/emulation/qemu-armv7/fixed-dtfs.dts [hunk 1]
```diff
@@ -14,56 +14,56 @@
...
                 offset = <0x0>;
-                size = <0x80000>; // 512KiB
+                size = <0x880000>; // 8.5MiB
                 file = "$(TARGET_DIR)/bootblob.bin";
             };
             area@1 {
                 description = "Fixed DTFS";
                 compatible = "ore-dtfs";
-                offset = <0x80000>;
+                offset = <0x880000>;
```
Comment (fotonick) [inline src/mainboard/emulation/qemu-armv7/fixed-dtfs.dts:23]: Could omit all of these `offset` entries if you wanted to; IIRC they're auto-computed. I'm not sure that being explicit here actually adds any value to a human reader.

File: src/mainboard/emulation/qemu-armv7/src/main.rs [hunk 2]
```diff
@@ -24,9 +23,10 @@ pub extern "C" fn _start() -> ! {
     uart_driver.pwrite(b"Welcome to oreboot\r\n", 0).unwrap();
     let s = &mut [uart_driver];
     let console = &mut DoD::new(s);
+    let mut w = print::WriteTo::new(console);
+    let spi = SliceReader::new(DTB);
 
-    if let Err(err) = print_fdt(console) {
-        let mut w = print::WriteTo::new(console);
+    if let Err(err) = print_fdt(&spi, &mut w) {
         write!(w, "error: {}\n", err).expect(err);
```
Comment (fotonick) [inline src/mainboard/emulation/qemu-armv7/src/main.rs:30]: I'm surprised that clippy isn't suggesting `writeln!`.

**Review by fotonick — DISMISSED** [review]
These net negative patches are so satisfying. Thanks for all the good work.

**Review by rjoleary — APPROVED** [review]

File: src/mainboard/emulation/qemu-armv7/fixed-dtfs.dts (hunk 1 above)
Comment (doriath) [reply src/mainboard/emulation/qemu-armv7/fixed-dtfs.dts:23]: Good point, removed the offset.

File: src/mainboard/emulation/qemu-armv7/src/main.rs (hunk 2 above)
Comment (doriath) [reply src/mainboard/emulation/qemu-armv7/src/main.rs:30]: Clippy is not enabled on all crates yet. There is pretty long list of BROKEN_CRATES_TO_CLIPPY in the Makefile (I am starting with #319), which is why this was not caught.

**Review by fotonick — APPROVED** [review]

**Pull request merged**